### ISSUE TITLE
Unable to determine poetry's path on poetry self:update --preview

### ISSUE THREAD

**Issue by batisteo** [sdispater/poetry#80]
I just tried `poetry` on Fedora 28 with Python 3.6.5.

```bash
$ pip3 install --user poetry
[successful output on next comment]

$ poetry self:update
You are using the latest version

$ poetry self:update --preview
Updating to 0.9.0a0
                                     
  [RuntimeError]                     
  Unable to determine poetry’s path  

self:update [--preview] [--] [<version>]

$ which poetry
~/.local/bin/poetry
```

Somewhat related to #53

**Comment by batisteo** [comment]
```bash
$ pip3 install --user poetry
Collecting poetry
  Downloading https://files.pythonhosted.org/packages/3a/c7/b8e7a8689b54ffb8d3f65cc4d021ee7ed47834cd73979d89ada1230ee8ff/poetry-0.8.6-py2.py3-none-any.whl (576kB)
    100% |████████████████████████████████| 583kB 1.9MB/s 
Requirement already satisfied: requests<3.0.0.0,>=2.18.0.0 in /usr/lib/python3.6/site-packages (from poetry)
Collecting pyrsistent<0.15.0.0,>=0.14.2.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/94/9e/bd3bd92d14cad2776ff558dfbcb0e591703240b7b55c230ac16754df5bf0/pyrsistent-0.14.2.tar.gz (97kB)
    100% |████████████████████████████████| 102kB 6.1MB/s 
Collecting cachecontrol<0.13.0.0,>=0.12.4.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/98/f5/76619a63f0e4a1d2f5a1792ebc233a395c648c63d3461dc0331479ef120a/CacheControl-0.12.4.tar.gz
Collecting requests-toolbelt<0.9.0.0,>=0.8.0.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/97/8a/d710f792d6f6ecc089c5e55b66e66c3f2f35516a1ede5a8f54c13350ffb0/requests_toolbelt-0.8.0-py2.py3-none-any.whl (54kB)
    100% |████████████████████████████████| 61kB 6.9MB/s 
Collecting pip-tools<3.0.0.0,>=2.0.0.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/4c/69/4f33fea6c50ff873b7905abc404d8c26aab2accf8c099d67a35aebc09871/pip_tools-2.0.2-py2.py3-none-any.whl (1.4MB)
    100% |████████████████████████████████| 1.5MB 857kB/s 
Collecting cleo<0.7.0.0,>=0.6.0.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/e5/a2/9e188fc717b80b9d85182a843234ee5d9fc8f208801f6db9d5cfcef60f15/cleo-0.6.5-py2.py3-none-any.whl (262kB)
    100% |████████████████████████████████| 266kB 4.2MB/s 
Collecting pkginfo<2.0.0.0,>=1.4.0.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/a3/fe/f32a48d48f40a7209be9825fba2566cab92364787cf37de2e08300dd6ce7/pkginfo-1.4.2-py2.py3-none-any.whl
Collecting msgpack-python<0.6.0.0,>=0.5.0.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/8a/20/6eca772d1a5830336f84aca1d8198e5a3f4715cd1c7fc36d3cc7f7185091/msgpack-python-0.5.6.tar.gz (138kB)
    100% |████████████████████████████████| 143kB 4.7MB/s 
Collecting pyparsing<3.0.0.0,>=2.2.0.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/6a/8a/718fd7d3458f9fab8e67186b00abdd345b639976bc7fb3ae722e1b026a50/pyparsing-2.2.0-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 4.6MB/s 
Collecting cachy<0.2.0.0,>=0.1.1.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/a8/20/4a8184ec140b0345fe3299d30945dd19df2ee4e129053f97c2c4709e32a2/cachy-0.1.1-py2.py3-none-any.whl
Collecting toml<0.10.0.0,>=0.9.0.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/f5/f9/044110c267e6408013b85166a7cfcd352cf85275aa8ce700aa5c0eb407ba/toml-0.9.4.tar.gz
Collecting lockfile<0.13.0.0,>=0.12.0.0 (from poetry)
  Downloading https://files.pythonhosted.org/packages/c8/22/9460e311f340cb62d26a38c419b1381b8593b0bb6b5d1f056938b086d362/lockfile-0.12.2-py2.py3-none-any.whl
Requirement already satisfied: jsonschema<3.0.0.0,>=2.6.0.0 in /usr/lib/python3.6/site-packages (from poetry)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python3.6/site-packages (from requests<3.0.0.0,>=2.18.0.0->poetry)
Requirement already satisfied: idna<2.7,>=2.5 in ./.local/lib/python3.6/site-packages (from requests<3.0.0.0,>=2.18.0.0->poetry)
Requirement already satisfied: urllib3<1.23,>=1.21.1 in /usr/lib/python3.6/site-packages (from requests<3.0.0.0,>=2.18.0.0->poetry)
Requirement already satisfied: six in /usr/lib/python3.6/site-packages (from pyrsistent<0.15.0.0,>=0.14.2.0->poetry)
Requirement already satisfied: click>=6 in ./.local/lib/python3.6/site-packages (from pip-tools<3.0.0.0,>=2.0.0.0->poetry)
Collecting first (from pip-tools<3.0.0.0,>=2.0.0.0->poetry)
  Downloading https://files.pythonhosted.org/packages/71/12/1b24a85f543658804027de982f7e5cb80543ea33ec396c887c099c75274c/first-2.0.1-py2.py3-none-any.whl
Collecting pylev<2.0.0.0,>=1.3.0.0 (from cleo<0.7.0.0,>=0.6.0.0->poetry)
  Downloading https://files.pythonhosted.org/packages/40/1c/7dff1d242bf1e19f9c6202f0ba4e6fd18cc7ecb8bc85b17b2d16c806e228/pylev-1.3.0-py2.py3-none-any.whl
Collecting pastel<0.2.0.0,>=0.1.0.0 (from cleo<0.7.0.0,>=0.6.0.0->poetry)
  Downloading https://files.pythonhosted.org/packages/9b/7e/7d701686013c0d7dae62e0977467232a6adc2e562c23878eb3cd4f97d02e/pastel-0.1.0-py3-none-any.whl
Installing collected packages: pyrsistent, msgpack-python, cachecontrol, requests-toolbelt, first, pip-tools, pylev, pastel, cleo, pkginfo, pyparsing, cachy, toml, lockfile, poetry
  Found existing installation: pyrsistent 0.11.13
    Uninstalling pyrsistent-0.11.13:
      Successfully uninstalled pyrsistent-0.11.13
  Running setup.py install for pyrsistent ... done
  Running setup.py install for msgpack-python ... done
  Running setup.py install for cachecontrol ... done
  Running setup.py install for toml ... done
Successfully installed cachecontrol-0.12.4 cachy-0.1.1 cleo-0.6.5 first-2.0.1 lockfile-0.12.2 msgpack-python-0.5.6 pastel-0.1.0 pip-tools-2.0.2 pkginfo-1.4.2 poetry-0.8.6 pylev-1.3.0 pyparsing-2.2.0 pyrsistent-0.14.2 requests-toolbelt-0.8.0 toml-0.9.4

You are using pip version 9.0.3, however version 10.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

$
```

**Comment by jgirardet** [comment]
Try update pip. Toml support start @ pip 10

**Comment by batisteo** [comment]
Yeah… but I won’t upgrade my system pip.
I’ll try in another venv then. Thanks for the info.

**Comment by sdispater** [comment]
@batisteo It's a known issue when `poetry` is installed with `pip --user` that's why this is not the recommended way of installing poetry.

Anyway, I am working on a solution and I'll let you know if I come up with something.

**Comment by sdispater** [comment]
Commit 5ac9794 on the `develop` branch will likely fix the issue. It will be available in the next `0.9.0` release.

**Comment by batisteo** [comment]
Applying these changes on my installed `0.8.6` fixed the issue :tada: 
Thanks!
